runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowMetrics.java

```diff
@@ -249,14 +249,33 @@ private boolean isMetricTentative(MetricUpdate metricUpdate) {
      */
     private MetricKey getMetricHashKey(MetricUpdate metricUpdate) {
       String fullStepName = metricUpdate.getName().getContext().get("step");
-      if (dataflowPipelineJob.transformStepNames == null
-          || !dataflowPipelineJob.transformStepNames.inverse().containsKey(fullStepName)) {
-        // If we can't translate internal step names to user step names, we just skip them
-        // altogether.
-        return null;
+
+      if (dataflowPipelineJob.getPipelineProto() != null
```

Comment:
nit: Might be readable to separate the conditionals to a separate method `getUniqueName` so that each of the `fullStepName = ...` lines can just be a `return`

Reply:
readability nit to avoid the pattern of setting a variable in a couple conditional blocks and then using it below the statement

Reply:
Refactored code.